daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/PackedBinaryTraits.scala

```diff
@@ -70,11 +71,16 @@ abstract class PackedBinaryDecimalBaseParser(
 
 abstract class PackedBinaryIntegerBaseParser(
   override val context: ElementRuntimeData,
-  signed: Boolean = false
 ) extends PrimParser
   with PackedBinaryConversion {
   override lazy val runtimeDependencies = Vector()
 
+  val signed = {
+    context.optPrimType.get match {
+      case n: NodeInfo.PrimType.PrimNumeric => n.isSigned
+      case _ => false
```

Comment:
I'm surprised the default case has coverage. This doesn't seem possible to me. A packed binary number should always have a numeric type. I wonder if codecov just can't detect this correctly?

Reply:
Also, this makes me concern the sign change wasn't correct? Previously we passed in `false` to a handful of these packed parsers, regardless of the type. I wonder if we don't allow unsigned primitives for those packed parsers, so using primType.isSigned is functionally the same?

Reply:
So this is used by non-PrimNumeric types like date/time via ConvertZonedCombinator...so that's why this has coverage

Reply:
Can you add a comment so that's clear in the future?